app/compare/diff_preprocessing.tsx

```diff
@@ -0,0 +1,135 @@
+  );
+  // Sort structured command lines so canonical always comes before original.
+  sortByProperty(invocation.structuredCommandLine, (commandLine) => commandLine.commandLineLabel);
+
+  if (hideTimingData) {
+    for (const commandLine of invocation.structuredCommandLine) {
+      removeTimingData(commandLine);
+    }
+  }
+
+  for (const event of invocation.event) {
+    if (sortEvents) {
+      delete event.sequenceNumber;
+    }
+
+    const buildEvent = event?.buildEvent;
+    const id = buildEvent?.id;
+    if (!id) continue;
+
+    if (id?.configuration) {
+      // The "makeVariable" map sometimes shows the same data but rendered in a different order;
+      // sorting the maps solves this problem.
+      sortEntriesByKey(buildEvent.configuration.makeVariable);
+    }
+
+    if (hideTimingData) {
+      delete event.eventTime;
+
+      if (id?.workspaceStatus) {
+        const timestampItem = buildEvent.workspaceStatus.item.find((item: any) => item.key === "BUILD_TIMESTAMP");
+        if (timestampItem) {
+          delete timestampItem.value;
+        }
+      } else if (id?.buildToolLogs) {
+        buildEvent.buildToolLogs.log = event.buildEvent.buildToolLogs.log.filter(
+          (log) => !["elapsed time", "critical path", "process stats"].includes(log.name)
+        );
+      } else if (id?.structuredCommandLine) {
+        removeTimingData(buildEvent.structuredCommandLine);
+      } else if (id?.buildMetrics) {
+        delete buildEvent.buildMetrics.timingMetrics;
+      } else if (id?.started) {
+        delete buildEvent.started.startTimeMillis;
+      } else if (id?.buildFinished) {
+        delete buildEvent.finished.finishTimeMillis;
+      }
+    }
+
+    if (hideUuids) {
+      if (id?.started) {
+        delete buildEvent.started.uuid;
+      }
+    }
+  }
+  return invocation;
+}
+
+function removeTimingData(commandLine: command_line.ICommandLine) {
+  for (const section of commandLine.sections) {
+    if (!section?.optionList?.option) continue;
+    section.optionList.option = section.optionList.option.filter((option) => option.optionName !== "startup_time");
+  }
+}
+
+function sortByProperty<T>(items: T[], property: (item: T) => any) {
+  items.sort((itemA, itemB) => {
+    const a = property(itemA);
+    const b = property(itemB);
+    return a < b ? -1 : a > b ? 1 : 0;
+  });
+}
+
+function sortEntriesByKey(object: Record<string, any>) {
+  // In ES6, object entries are sorted by the order in which they're inserted.
+  // So, deleting all entries and re-inserting them in sorted order effectively
+  // sorts the object.
+  for (const key of Object.keys(object).sort()) {
+    const value = object[key];
+    delete object[key];
+    object[key] = value;
+  }
+}
+
+function replaceAll(str: string, value: string, replacement: string) {
+  return str.split(value).join(replacement);
+}
+
+function replaceAllJsonValues(json: string, keys: string[], replacement: string) {
```

Comment:
is this still used? same with `replaceAll` above

Reply:
Nope, removed